scimath_f/lawson.f

```diff
@@ -824,6 +872,7 @@ SUBROUTINE DLDP (G,MDG,M,N,H,X,XNORM,W,INDEX,MODE)
       SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
       implicit double precision (a-h,o-z)
       DIMENSION A(MDA,N), B(1), X(1), W(1), ZZ(1)
+      DOUBLE PRECISION DUMMY(1)
```

Comment:
Why is this here?

Reply:
It's required to define to give it 'the right shape'. If I remember well: Old Fortran allowed undefined use of Dummy, and made it a scalar. Old fortran than accepted that scalar in a function call and convert it to a one-dim array.